attestation/src/measurement.rs

```diff
@@ -78,15 +58,90 @@ enum CsrCdi {}
 #[derive(Clone, Debug)]
 enum SealingCdi {}
 
-const TCG_PCR_MAPPING: [u8; MSMT_REGISTERS] = [
-    0,  // TCG PCR 0  - SRTM
-    1,  // TCG PCR 1  - Platform config
-    8,  // TCG PCR 8  - OS
-    9,  // TCG PCR 9  - OS
-    17, // TCG PCR 17 - DRTM
-    18, // TCG PCR 18 - DRTM
-    19, // TCG PCR 19 - DRTM
-    20, // TCG PCR 20 - DRTM
+struct MeasurementRegisterBuilder {
+    // The DiceTcbInfo FWIDs list index.
+    fwid_index: u8,
+
+    // TCG PCR index.
+    pcr_index: u8,
+
+    // A static register can not be extended after the platform is booted, i.e.
+    // after the static TCB is finalized.
+    // A call to `finalize` will lock static registers, making subsequent calls
+    // to `extend` fail.
+    static_measurement: bool,
+
+    // For a given workload, a stable measurement register must not change
+    // between boots. If it does change, this means the workload TCB actually
+    // changed. For example, a platform config measurement is not stable, while
+    // an OS one is.
```

Comment:
Nit: Consider rewording without the trailing "is". Perhaps: Examples of stable measurements include <blah>, but <blah> is non-stable.